File: 01_Language/01_Functions/php/date_offset_get.php
<?php

$winter = new DateTime('2010-12-21', new DateTimeZone('America/New_York'));
$summer = new DateTime('2008-06-21', new DateTimeZone('America/New_York'));

echo $winter->getOffset() . "\n";
echo $summer->getOffset() . "\n";

$winter = date_create('2010-12-21', timezone_open('America/New_York'));
$summer = date_create('2008-06-21', timezone_open('America/New_York'));

echo date_offset_get($winter) . "\n";
echo date_offset_get($summer) . "\n";

File: 01_Language/01_Functions/python/date_offset_get.py
# coding: utf-8

import pytz
import datetime

if __name__ == '__main__':
    tz = pytz.timezone("America/New_York")
    winter = datetime.datetime.strptime("2010-12-21", "%Y-%m-%d")
    summer = datetime.datetime.strptime("2008-06-21", "%Y-%m-%d")
    print(tz.utcoffset(winter).total_seconds())
    print(tz.utcoffset(summer).total_seconds())
